libs/core-ui/src/lib/Cohort/ManualCohortManagement/CohortEditorFilter.tsx

```diff
@@ -42,47 +51,83 @@ export interface ICohortEditorFilterProps {
   setComparison(ev?: React.FormEvent<IComboBox>, item?: IComboBoxOption): void;
   setNumericValue(
     delta: number,
-    column: IJointMeta,
     index: number,
-    stringVal: string
+    stringVal: string,
+    range?: INumericRange
   ): string | void;
   saveState(index?: number): void;
   cancelFilter(): void;
 }
+const maxLength = 18;
 export class CohortEditorFilter extends React.Component<ICohortEditorFilterProps> {
-  private readonly dataArray: IComboBoxOption[] = new Array(
-    this.props.jointDataset.datasetFeatureCount
-  )
-    .fill(0)
-    .map((_, index) => {
-      const key = JointDataset.DataLabelRoot + index.toString();
-      return {
-        key,
-        text: this.props.jointDataset.metaDict[key].abbridgedLabel
-      };
-    });
+  private readonly dataArray: IComboBoxOption[] = this.props.isRefactorFlightOn
+    ? this.props.dataset.feature_names.map((featureName, index) => {
+        return {
+          index,
+          key: featureName,
+          text:
+            featureName.length <= maxLength
+              ? featureName
+              : `${featureName.slice(0, maxLength)}...`
+        };
+      })
+    : new Array(this.props.jointDataset.datasetFeatureCount)
+        .fill(0)
+        .map((_, index) => {
+          const key = JointDataset.DataLabelRoot + index.toString();
+          return {
+            index,
+            key,
+            text: this.props.jointDataset.metaDict[key].abbridgedLabel
+          };
+        });
+
   public render(): React.ReactNode {
     const selectedMeta =
-      this.props.jointDataset.metaDict[this.props.openedFilter.column];
```

Comment:
because you're checking the refactor flight a few times in the render function, would it make sense to call a different render function whether the flight is on or not ? something like : `public render(): React.ReactNode { if(this.props.isRefactorFlightOn) {
renderWithFlight();}
else {
render();
}
}`